src/Kiota.Builder/CodeDOM/IKindableElement.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Kiota.Builder.CodeDOM;
+
+public interface IKindableElement<TKind> where TKind : Enum
```

Comment:
Does this mean we should probably clean up the `CodeTerminalWithKind` usages as well? They do look to do the same thing and probably can be dropped.

Reply:
ugh... I forgot we had this thing. Deleting this interface since the base abstract class implements some behavior for us, reducing code duplication.